src/python/pants/backend/python/packaging/pyoxidizer/rules_integration_test.py

```diff
@@ -44,4 +68,30 @@ def test_end_to_end() -> None:
         # Check that the binary is executable.
         bin_path = next(Path("dist", f"{tmpdir}.hellotest", "bin").glob("*/debug/install/bin"))
         bin_stdout = subprocess.run([bin_path], check=True, stdout=subprocess.PIPE).stdout
-        assert bin_stdout == b"Hello test\n"
+        assert bin_stdout == b"Hello world!\n"
+
+
+def test_requires_wheels() -> None:
```

Comment:
This test is what prompted making building the PEX lazier. I didn't want to waste time in this test :) Now we validate `dependencies` are correct before building the PEX.